mono/metadata/profiler-events.h

```diff
@@ -93,8 +93,8 @@ MONO_PROFILER_EVENT_0(gc_finalizing, GCFinalizing)
 MONO_PROFILER_EVENT_0(gc_finalized, GCFinalized)
 MONO_PROFILER_EVENT_1(gc_finalizing_object, GCFinalizingObject, MonoObject *, object)
 MONO_PROFILER_EVENT_1(gc_finalized_object, GCFinalizedObject, MonoObject *, object)
-MONO_PROFILER_EVENT_5(gc_root_register, RootRegister, const mono_byte *, start, uintptr_t, size, MonoGCRootSource, source, const void *, key, const char *, name)
```

Comment:
These are public API, you can't change their signatures.

Reply:
Even in a compatible easier to use way? It is far fetched that this will break any source or binaries, changing foo* to void*.

Reply:
```bash
bash ~/stuff $ cat test.c
void foo (char *p)
{
}

static void (*fnptr) (void *p);

int main (void)
{
        fnptr = foo;

        return 0;
}
bash ~/stuff $ gcc -Wall -Werror test.c
test.c: In function ‘main’:
test.c:9:8: error: assignment from incompatible pointer type [-Werror=incompatible-pointer-types]
  fnptr = foo;
        ^
cc1: all warnings being treated as errors
```

Reply:
Ok let me try to put it back. Should just be a bunch of casts to compensate.